backend/models/market.go

```diff
@@ -21,3 +21,21 @@ type Market struct {
 	CreatorUsername         string    `json:"creatorUsername" gorm:"not null"`
 	Creator                 User      `gorm:"foreignKey:CreatorUsername;references:Username"`
 }
+
+func (m *Market) IsClosed() bool {
+	if !m.IsResolved {
+		return false
+	}
+
+	// TODO: Decide if we should be using one or both of the below conditions
```

Comment:
Same comment here regarding TODO's in the code.

Please let me know if we should add some kind of convention regarding leaving TODO's in the code, e.g. within the README documentation.

My thought is, the less we put into the code base, even comments, the better.

Reply:
I generally agree, I think if we decide to leave a TODO in the code, it should have an issue linked to it, but in general it's better to avoid.

Some comments can be helpful though, as they allow us to generate api documentation via go doc: https://tip.golang.org/doc/comment

Reply:
Also, @pwdel based on your comments explaining the market fields, it sounds like we should be checking both Resolution and FinalResolution. Is this understanding correct?

Reply:
So here is the code I think we're talking about, right?

```
	market.IsResolved = true
	market.ResolutionResult = resolutionData.Outcome
	market.FinalResolutionDateTime = time.Now()
```

And in the data model: https://github.com/openpredictionmarkets/socialpredict/blob/main/backend/models/market.go

```
	ResolutionDateTime      time.Time `json:"resolutionDateTime" gorm:"not null"`
	FinalResolutionDateTime time.Time `json:"finalResolutionDateTime"`
```

What if we renamed `ResolutionDateTime` to, `ScheduledMarketCloseDateTime` and then we just don't touch it upon resolution? That way there is a record of when the scheduled market close time originally was (the time after which point no betting could be set), and so there is more transparency in terms of whether a market was resolved early, on time, or late.

Hypothetically, the `ScheduledMarketCloseDateTime` could be re-scheduled. If we needed to build ina changelog for this, e.g. showing when the scheduled Close datetime was changed, hence opening up the market for bets again and closing them again, this would be knowledge to anyone looking at the graph perhaps wondering why there was activity in dispersed intervals.

Hypothetically much of this could be covered in the comments, but if anyone decides not to use the comment features of the platform (and just use something like Discord to discuss markets), a hypothetical change log could suffice.

Reply:
I think renaming to `ScheduledCloseDateTime` makes sense. I think it's worth discussing some more about what type this should be if we're planning to allow market closing to be rescheduled. One possibility that jumps to mind is to have a slice of time values, but this loses information about when the changes are made; we could maybe work around this with a corresponding slice of times tracking when the updates are made. Another possibility is to consider using a db with support for time series data.

I'm concerned that updating the corresponding json field from `resolutionDateTime` to `scheduledCloseDateTime` could be a breaking change. Thoughts @markokovac16?

Reply:
> I'm concerned that updating the corresponding json field from `resolutionDateTime` to `scheduledCloseDateTime` could be a breaking change. Thoughts @markokovac16?

Not really, quick search and replace in the project sorts that out, frontend is not super complex yet so stuff like that is fairly easy to change. @ajlacey